lib/simulatelocation/index.js

```diff
@@ -0,0 +1,51 @@
+//We just need to write any data to the client. It doesn't matter what we send
+const SIMULATE_LOCATION_SERVICE_NAME = 'com.apple.dt.simulatelocation';
+
+const RESET_MESSAGE = Buffer.from([0, 0, 0, 1]);
+const SET_LOCATION_MESSAGE = Buffer.from([0, 0, 0, 0]);
+
+class SimulateLocationService {
+
+  constructor (socketClient) {
+    this._socketClient = socketClient;
+
+  }
+  /**
+   * Reset the mock location to the phones original settings
+   */
+  resetLocation () {
+    this._socketClient.write(RESET_MESSAGE);
+  }
+
+  /**
+   * Set the mock location on the device
+   * @param {string|number} lat The latitude that wants to be set on the device
+   * @param {string|number} long The longitude that wants to be set on the device
+   */
+  setLocation (lat, long) {
+    lat = lat.toString();
+    long = long.toString();
+
+    const lat_length = Buffer.alloc(4);
```

Comment:
this stuff can actually be done in a loop for both attitude and longitude

Reply:
It can but I think it looks more meaningful like this